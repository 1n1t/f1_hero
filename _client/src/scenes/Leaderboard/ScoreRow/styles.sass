.ScoreRow
  display: flex
  padding: 0 30px
  &:not(:last-child)
    margin-bottom: 20px

  .username
    flex-grow: 1
    margin-left: 20px
