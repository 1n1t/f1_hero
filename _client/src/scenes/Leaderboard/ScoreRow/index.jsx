import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

class ScoreRow extends Component {
	shouldComponentUpdate() {
		return false;
	}

	render() {
		const { index, user, total } = this.props;

		return (
			<div className="ScoreRow">
				<div>{index + 1}.</div>
				<div className="username">{user}</div>
				<div>{total}pt</div>
			</div>
		);
	}
}

ScoreRow.propTypes = {
	user: PropTypes.string.isRequired,
	total: PropTypes.number.isRequired,
	index: PropTypes.number.isRequired
};

export default ScoreRow;
