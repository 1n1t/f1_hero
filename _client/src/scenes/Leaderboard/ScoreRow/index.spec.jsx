import r from 'Root/jest/utils/renderer';
import leaderboardSample from 'JestClient/samples/leaderboard';
import ScoreRow from './index';

const defaultProps = {
	...leaderboardSample.scores[0],
	index: 0
};

r.init(ScoreRow, defaultProps);

it('renders correctly', () => {
	const wrapper = r.render(null, true);
	expect(wrapper).toMatchSnapshot();
});
