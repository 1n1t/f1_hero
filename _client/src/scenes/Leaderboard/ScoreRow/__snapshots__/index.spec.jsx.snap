// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ScoreRow"
>
  <div>
    1
    .
  </div>
  <div
    className="username"
  >
    John Doe
  </div>
  <div>
    123
    pt
  </div>
</div>
`;
